src/walletSdk/horizon/Account.ts

```diff
@@ -0,0 +1,52 @@
+import { Keypair, Transaction, FeeBumpTransaction } from "stellar-sdk";
+
+class AccountKeypair {
+  keypair: Keypair;
+  constructor(keypair: Keypair) {
+    this.keypair = keypair;
+  }
+
+  get publicKey(): string {
+    return this.keypair.publicKey();
+  }
+
+  toString(): string {
+    return `${this.constructor.name}(address=${this.publicKey})`;
+  }
+}
+
+export class PublicKeypair extends AccountKeypair {
+  keypair: Keypair;
+  constructor(keypair: Keypair) {
+    super(keypair);
+  }
+}
+
+export class SigningKeypair extends AccountKeypair {
+  keypair: Keypair;
+  constructor(keypair: Keypair) {
+    if (!keypair.canSign()) {
+      throw new Error("This keypair doesn't have a private key and can't sign");
+    }
+    super(keypair);
+  }
+
+  static fromSecret = (secret: string): SigningKeypair => {
+    return new SigningKeypair(Keypair.fromSecret(secret));
+  };
+
+  get secretKey(): string {
+    return this.keypair.secret();
+  }
+
+  sign(
+    transaction: Transaction | FeeBumpTransaction
+  ): Transaction | FeeBumpTransaction {
+    transaction.sign(this.keypair);
+    return transaction;
+  }
+
+  toString(): string {
+    return `${this.constructor.name}(address=${this.publicKey})`;
+  }
+}
```

Comment:
similarly I didn't try to extend the String class like kotlin does [here](https://github.com/stellar/kotlin-wallet-sdk/blob/e6c801e24e7c716b876a67098131dd0b4d2f1691/wallet-sdk/src/main/kotlin/org/stellar/walletsdk/horizon/Account.kt#L50)

imo extending native classes may lead to some wonky behavior, but happy to discuss

Reply:
Yeah adding method `PublicKeyPair.fromString()` should be good enough

Reply:
https://github.com/stellar/typescript-wallet-sdk/pull/21/commits/2f244a59a2922afac45f7e3cb9aaeded5d062a72